### PR TITLE
feat: [FFM-9037]: Change waiting for cleanup to appear on reason not status

### PR DESCRIPTION
**Pull request by laura-mannus**
### Summary

After chatting with Andy I realised that I had misunderstood the requirements, the 'Waiting for Cleanup' label should appear if the potentially stale flag is "marked_by_user" as the reason, not just stale. If we displayed it only on the stale attribute it would always show.

I changed this logic so it works as expected, and added in the correct enums.

#### Screenshots

image

`{
  "itemCount": 4,
  "pageCount": 1,
  "pageIndex": 0,
  "pageSize": 15,
  "featureCounts": {
    "totalActive": 0,
    "totalArchived": 0,
    "totalEnabled": 0,
    "totalFeatures": 4,
    "totalPermanent": 0,
    "totalPotentiallyStale": 4,
    "totalRecentlyAccessed": 0
  },
  "features": [
    {
      "archived": false,
      "createdAt": 1688039764343,
      "defaultOffVariation": "false",
      "defaultOnVariation": "false",
      "description": "",
      "envProperties": {
        "defaultServe": {
          "variation": "false"
        },
        "environment": "env1",
        "jiraEnabled": false,
        "modifiedAt": 1688039764355,
        "offVariation": "false",
        "pipelineConfigured": false,
        "rules": null,
        "state": "off",
        "variationMap": null,
        "version": 1
      },
      "evaluation": "",
      "evaluationIdentifier": "",
      "identifier": "stale4",
      "kind": "boolean",
      "modifiedAt": 1688039764343,
      "name": "stale4",
      "owner": [
        "laura.mannus@harness.io"
      ],
      "permanent": false,
      "prerequisites": null,
      "project": "laurastaleproj",
      "services": [],
      "stale": true,
      "staleReason": "marked_by_user",
      "tags": [],
      "variations": [
        {
          "identifier": "true",
          "name": "True",
          "value": "true"
        },
        {
          "identifier": "false",
          "name": "False",
          "value": "false"
        }
      ]
    },
    {
      "archived": false,
      "createdAt": 1688039751237,
      "defaultOffVariation": "false",
      "defaultOnVariation": "false",
      "description": "",
      "envProperties": {
        "defaultServe": {
          "variation": "false"
        },
        "environment": "env1",
        "jiraEnabled": false,
        "modifiedAt": 1688039751249,
        "offVariation": "false",
        "pipelineConfigured": false,
        "rules": null,
        "state": "off",
        "variationMap": null,
        "version": 1
      },
      "evaluation": "",
      "evaluationIdentifier": "",
      "identifier": "stale3",
      "kind": "boolean",
      "modifiedAt": 1688039751237,
      "name": "stale3",
      "owner": [
        "laura.mannus@harness.io"
      ],
      "permanent": false,
      "prerequisites": null,
      "project": "laurastaleproj",
      "services": [],
      "stale": true,
      "staleReason": "matched_criteria",
      "tags": [],
      "variations": [
        {
          "identifier": "true",
          "name": "True",
          "value": "true"
        },
        {
          "identifier": "false",
          "name": "False",
          "value": "false"
        }
      ]
    },
    {
      "archived": false,
      "createdAt": 1688039739026,
      "defaultOffVariation": "false",
      "defaultOnVariation": "false",
      "description": "",
      "envProperties": {
        "defaultServe": {
          "variation": "false"
        },
        "environment": "env1",
        "jiraEnabled": false,
        "modifiedAt": 1688039739042,
        "offVariation": "false",
        "pipelineConfigured": false,
        "rules": null,
        "state": "off",
        "variationMap": null,
        "version": 1
      },
      "evaluation": "",
      "evaluationIdentifier": "",
      "identifier": "stale2",
      "kind": "boolean",
      "modifiedAt": 1688039739026,
      "name": "stale2",
      "owner": [
        "laura.mannus@harness.io"
      ],
      "permanent": false,
      "prerequisites": null,
      "project": "laurastaleproj",
      "services": [],
      "stale": true,
      "staleReason": "matched_criteria",
      "tags": [],
      "variations": [
        {
          "identifier": "true",
          "name": "True",
          "value": "true"
        },
        {
          "identifier": "false",
          "name": "False",
          "value": "false"
        }
      ]
    },
    {
      "archived": false,
      "createdAt": 1685361286002,
      "defaultOffVariation": "false",
      "defaultOnVariation": "false",
      "description": "",
      "envProperties": {
        "defaultServe": {
          "variation": "false"
        },
        "environment": "env1",
        "jiraEnabled": false,
        "modifiedAt": 1688039698432,
        "offVariation": "false",
        "pipelineConfigured": false,
        "rules": null,
        "state": "off",
        "variationMap": null,
        "version": 1
      },
      "evaluation": "",
      "evaluationIdentifier": "",
      "identifier": "stale1",
      "kind": "boolean",
      "modifiedAt": 1688039686002,
      "name": "stale1",
      "owner": [
        "laura.mannus@harness.io"
      ],
      "permanent": false,
      "prerequisites": null,
      "project": "laurastaleproj",
      "services": [],
      "stale": true,
      "staleReason": "matched_criteria",
      "tags": [],
      "variations": [
        {
          "identifier": "true",
          "name": "True",
          "value": "true"
        },
        {
          "identifier": "false",
          "name": "False",
          "value": "false"
        }
      ]
    }
  ]
}`

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cf/pages/feature-flags/FlagStatus.tsx [hunk 1]
```diff
@@ -22,19 +22,27 @@ export enum FeatureFlagStatus {
   RECENTLY_ACCESSED = 'recently-accessed',
   ARCHIVED = 'archived'
 }
+
+export enum StaleFlagStatusReason {
+  WAITING_FOR_CLEANUP = 'marked_by_user',
+  POTENTIALLY_STALE = 'matched_criteria'
+}
```
Comment (laura-mannus) [inline src/modules/75-cf/pages/feature-flags/FlagStatus.tsx:29]: I know these don't match and aren't ideal, they are not named appropriately in the backend, so for readability this is why they don't match

File: src/modules/75-cf/pages/feature-flags/FlagStatus.tsx [hunk 2]
```diff
@@ -22,19 +22,27 @@ export enum FeatureFlagStatus {
...
 
-export const FlagStatus: React.FC<FlagStatusProps> = ({ status, lastAccess, stale }) => {
+export const FlagStatus: React.FC<FlagStatusProps> = ({ status, lastAccess, staleReason }) => {
   const { getString } = useStrings()
   const isNeverRequested = status === FeatureFlagStatus.NEVER_REQUESTED
   const isPotentiallyStale = status === FeatureFlagStatus.POTENTIALLY_STALE
 
   const flagCleanupEnabled = useFeatureFlag(FeatureFlag.FFM_8344_FLAG_CLEANUP)
 
+  const isWaitingForCleanup = staleReason == StaleFlagStatusReason.WAITING_FOR_CLEANUP
```
Comment (Kira-Jennings) [inline src/modules/75-cf/pages/feature-flags/FlagStatus.tsx:44]: Can you use strict comparison here?
```suggestion
  const isWaitingForCleanup = staleReason === StaleFlagStatusReason.WAITING_FOR_CLEANUP
```

Comment (laura-mannus) [reply src/modules/75-cf/pages/feature-flags/FlagStatus.tsx]: Thanks! Have updated

**Review by Kira-Jennings — APPROVED** [review]
Looks good!

**Review by crystal-fong — APPROVED** [review]

**Pull request merged**